### PR TITLE
Removed Pangolin

### PR DESCRIPTION
**Pull request by greenjava**
- removed dependency to Pangolin in ORB_SLAM2
- removed class MapDraver, Viewer and FrameDrawer
- removed deprecated examples ROS
- added options to disable build examples or utility

### CODE REVIEW

**Review by PamplemousseMR — APPROVED** [review]

**Review by schweitzer — APPROVED** [review]

File: CMakeLists.txt [hunk 1]
```diff
@@ -52,44 +51,41 @@ add_subdirectory(${PROJECT_SOURCE_DIR}/Thirdparty/DBoW2/)
...
-${EIGEN3_INCLUDE_DIR}
-${Pangolin_INCLUDE_DIR}
-${PROJECT_BINARY_DIR}
+    ${PROJECT_SOURCE_DIR}
+    ${PROJECT_SOURCE_DIR}/Thirdparty/DBoW2
+    ${PROJECT_SOURCE_DIR}/Thirdparty/g2o
+    ${PROJECT_SOURCE_DIR}/include
+    ${OpenCV_INCLUDE_DIRS}
+    ${EIGEN3_INCLUDE_DIR}
+    ${Pangolin_INCLUDE_DIR}
```
Comment (schweitzer) [inline CMakeLists.txt]: Why Pangolin_INCLUDE_DIR ?

File: CMakeLists.txt [hunk 2]
```diff
@@ -52,44 +51,41 @@ add_subdirectory(${PROJECT_SOURCE_DIR}/Thirdparty/DBoW2/)
...
 
 target_link_libraries(ORB_SLAM2
-${OpenCV_LIBS}
-${EIGEN3_LIBS}
-${Pangolin_LIBRARIES}
-DBoW2
-g2o
+    ${OpenCV_LIBS}
+    ${EIGEN3_LIBS}
+    ${Pangolin_LIBRARIES}
```
Comment (schweitzer) [inline CMakeLists.txt]: Same here you have removed Pangolin libraries.

**Review by schweitzer — CHANGES_REQUESTED** [review]

**Review by emiliehar — APPROVED** [review]

**Review by dweckmann — APPROVED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (greenjava) [reply CMakeLists.txt]: oups, fixed in 14a723d783aa68df657814a0c75d9858909f8d06

File: CMakeLists.txt (hunk 2 above)
Comment (greenjava) [reply CMakeLists.txt]: oups, fixed in 14a723d783aa68df657814a0c75d9858909f8d06

**Review by schweitzer — APPROVED** [review]

**Pull request merged**